tdms/include/interpolate_Hfield.h

```diff
@@ -1,18 +1,48 @@
-/* Interpolation functions for each of the H-field components.
-Decided to do with 3 separate functions for readability; as this avoids switches with a lot of cases within functions, and the interpolation scheme might be different in each direction anyway, so there is no point having a centralised function for all of them.
-
-Additionally, generalised functions for performing the 2D interpolation is not pretty code to look at (although is theoretically possible if we wanted a slight speedup).
-This forces us to pass in I,J,K to _every_ component function, rather than only the number of cells in the appropriate direction.
-
-If we want to interpolate all the H-field components, there are functions for this which call the individual component functions separately.
-*/
-
+/**
+ * @brief Interpolate the x-component of the magnetic field to the centre of Yee cell i,j,k
+ *
+ * Interpolation for the magnetic field must be done across two dimensions - first we must interpolate to obtain nearby field values parallel to the centre of the Yee cell, then we must interpolate again in another dimension to the centre of the Yee cell itself.
```

Comment:
maybe worth setting a line length limit in your IDE. although we can just clang-format everything in future

Reply:
For some reason VSCode defaults this to 200 :sweat_smile: - have reduced it to a more managable 100 now